blog/docs/articles/Building-Stateful-applications-with-Knative-and-Restate.md

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return "", err
+	}
+
+	// Prepare an awakeable to await the email activation
+	awakeable := restate.Awakeable[restate.Void](ctx)
+
+	// Send the activation email
+	_, err = restate.Run[restate.Void](ctx, func(ctx restate.RunContext) (restate.Void, error) {
+		return restate.Void{}, sendEmail(newUser.Username, awakeable.Id())
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Await the activation
+	_, err = awakeable.Result()
+	if err != nil {
+		return "", err
+	}
+
+	// Activate the user
+	_, err = restate.Object[restate.Void](ctx, "User", newUser.Username, "Activate").Request(user)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("The new user %s is signed up and activated", newUser.Username), nil
+}
+```
+
+Using `restate.Call` we can invoke other Restate services.
+These requests are guaranteed to be executed exactly once.
+
+With `restate.Awakeable` we can await an arbitrary event happening.
+You can complete requests simply [sending HTTP requests](https://docs.restate.dev/develop/ts/awakeables#completing-awakeables) to Restate providing the Awakeable id.
+In our example, the email will embed a link containing the Awakeable id, which will be completed once the user clicks on the verification button.
+
+With `restate.Run` we can execute any arbitrary piece of code and memoize the result, such that in case of a crash, Restate won't re-execute that chunk of code, but will load the stored result and use it for the subsequent operations.
+
+### Start the HTTP service and deploy it with Knative
+
+To expose the services using HTTP:
+
+```golang
+func main() {
+	// Read PORT env injected by Knative Serving
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "9080"
+	}
+	bindAddress := fmt.Sprintf(":%s", port)
+
+	// Bind services to the Restate HTTP/2 server
+	srv := server.NewRestate().
+		Bind(restate.Reflect(&userObject{})).
+		Bind(restate.Reflect(&signupService{}))
+
+	// Start HTTP/2 server
+	if err := srv.Start(context.Background(), bindAddress); err != nil {
+		slog.Error("application exited unexpectedly", "err", err.Error())
+		os.Exit(1)
+	}
+}
+```
+
+You can now build the container image using your tools, e.g. with `ko`:
+
+```shell
+$ ko build main.go -B
+```
+
+And deploy it with `kn`:
+
+```shell
+$ kn service create signup \
+  --image $MY_IMAGE_REGISTRY/main.go \
+  --port h2c:8080
+```
+
+Before sending requests, you need to tell Restate about your new service deployment:
+
+```shell
+$ restate deployments register http://signup.default.svc
+```
+
+And this is it! You're now ready to send requests:
+
+```shell
+$ curl http://localhost:8080/Signup/Signup --json '{"username": "slinkydeveloper", "name": "Francesco", "surname": "Guardiani", "password": "Pizza-without-pineapple"}'
+```
+
+Please note: some parts of the code example are omitted for brevity, check the [full example](TODO) for more details and how to run this locally with `kind`.
```

Comment:
Right now i'm thinking to publish this in the restate examples repo, but if you want i can also publish it somewhere else (e.g. in the examples in this repo?)

Reply:
Added a link to restate's examples repo.